pkg/collector/hotswap.go

```diff
@@ -19,35 +19,52 @@ import (
 	"sync"
 )
 
-// A HotSwap wraps a http.Handler and adds the support for a new handler to
+// HandlerCloser is an interface for a http.Handler that processes data
+// asynchronously and therefore must be closed. Once Close is called,
+// the caller must ensure that no further calls to ServeHTTP are made.
+type HandlerCloser interface {
+	Close()
+	ServeHTTP(http.ResponseWriter, *http.Request)
```

Comment:
Does it work to embed the interface here?

```
type HandlerCloser interface {
  Close()
  http.Handler
}
```

Reply:
It does! That's definitely a better choice, thanks.